transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionEntity.java

```diff
@@ -133,6 +135,32 @@ private Behavior paymentPending(TransactionState state) {
     private Behavior paymentSubmitted(TransactionState state) {
         BehaviorBuilder builder = newBehaviorBuilder(state);
 
+        builder.setCommandHandler(SubmitPaymentStatus.class, (cmd, ctx) -> {
+            if (cmd.getUserId().equals(state().getTransaction().get().getCreator())) {
+                if(cmd.isApproved())
+                    return ctx.thenPersist(new PaymentApproved(entityUUID()), (e) -> ctx.reply(Done.getInstance()));
+                else
+                    return ctx.thenPersist(new PaymentRejected(entityUUID()), (e) -> ctx.reply(Done.getInstance()));
+            } else
+                throw new Forbidden("Only the item creator can approve or reject payment");
+        });
+
+        builder.setEventHandlerChangingBehavior(PaymentApproved.class, evt ->
+                paymentConfirmed(state().withStatus(TransactionStatus.PAYMENT_CONFIRMED))
+        );
+
+        builder.setEventHandlerChangingBehavior(PaymentRejected.class, evt ->
+                paymentConfirmed(state().withStatus(TransactionStatus.PAYMENT_PENDING))
+        );
```

Comment:
@ignasi35, @yg-apaza and I discussed this on our call... here's what I was thinking: if the payment is rejected, that doesn't necessary end the whole thing. The seller and buyer can try again to negotiate payment, so it should return to the `paymentPending` behavior.

Eventually, we should add the option of the seller cancelling the transaction as a manual step.